### DISCUSSION TITLE
AnimationChannel does not work with AnimatedTexture

### DISCUSSION THREAD

**Discussion by jsFacled** [AlmasB/FXGL#1335] [General]
Question: AnimationChannel does not work with AnimatedTexture.
# I have an image with 4 rows (left,right,up,down), in each row there are 3 frames.
# I can extract a part of the image and save it with subTexture():
AnimatedTexture animTexture = texture("imageSprite.png").subTexture(new Rectangle2D(0, imagePlayerY, imagePlayer.getWidth(), 30)).toAnimatedTexture(3, Duration.seconds(0.6)).loop();
I thus obtain an AnimationTexture with the image cropped.
# The problem is that AnimationChannel does not work with AnimatedTexture.
# I have not been able to update the animatedTexture in playerComonent.
# So what I've done is edit the sprite.png and have 4 separate images.
# But the question is: how can I update animatedTexture using subTexture every time the movement is made (onKey(KeyCode.A, “left”, etc.). Since this way we would only be using a png resource to all movements.

**Comment by AlmasB** [comment]
Hi, you check the syntax in this [sample](https://github.com/AlmasB/FXGL/blob/9504e79c8dbe4e4fa7d56723afce0aea89606b58/fxgl-samples/src/main/java/intermediate/animation/AnimatedTextureSample.java)

See if it works for you and let me know if it doesn't.

**Reply by jsFacled** [reply]
Brilliant! It works perfectly using the full code in this example. Now I must implement it from my code, capturing the inputs, etc.
Because in the example class Component, EntityFactory, etc. is not used.
I was using the fxgl.texture api, but not fxgl.animation, interpolators, etc.
As soon as I get it to work (and if it doesn't too ) I'll get back to the conversation.
Thank you!!

**Comment by jsFacled** [comment]
I have managed to capture the 4 positions Up-Down-Right-Left from the same template. But I had to perform the construction of the Player every time the corresponding action is executed.
Question: is it inefficient to do it like this?
Here is the link to the code. [App3](https://github.com/jsFacled/javafxgl/blob/main/demoImageAnim/src/main/java/App3.java)

**Reply by AlmasB** [reply]
In most cases, yes, this is not efficient. Ideally, only one Entity is needed. Based on movements (up, down, etc.) you can then set the required view to the entity. Here's a quick example:

```
Image image = image("player.png");

animIdle = new AnimationChannel(image, 4, 32, 42, Duration.seconds(1), 1, 1);
animWalk = new AnimationChannel(image, 4, 32, 42, Duration.seconds(0.66), 0, 3);

texture = new AnimatedTexture(animIdle);
texture.loop();

//// set channel 1
texture.loopAnimationChannel(animWalk);

//// set channel 2
texture.loopAnimationChannel(animIdle);    
```

The idea is that each directional animation becomes a channel, which you can then set to the animated texture.

Hope this helps.
